### ISSUE TITLE
"From image" example results in stack overflow

### ISSUE THREAD

**Issue by RubberDuckShobe** [Aiving/material-colors#31]
Hello, I've tried testing this library using the "From image" example code from the readme, but the program seems to overflow its stack when calling ``ThemeBuilder::with_source``.

I've tried using an image smaller than the 1080p one that the example uses, but it still crashes.
```rust
use material_colors::{
    image::{FilterType, ImageReader},
    theme::ThemeBuilder,
};

#[tokio::main]
async fn main() -> Result<(), reqwest::Error> {
    let image = reqwest::get("https://picsum.photos/200/300")
        .await?
        .bytes()
        .await?
        .to_vec();

    let mut data = ImageReader::read(image).expect("failed to read image");

    // Lancsoz3 takes a little longer, but provides the best pixels for color extraction.
    // However, if you don't like the results, you can always try other FilterType values.
    data.resize(128, 128, FilterType::Nearest);

    let theme = ThemeBuilder::with_source(ImageReader::extract_color(&data)).build();

    // Do whatever you want...

    Ok(())
}
```

**Comment by Aiving** [comment]
I tried to reproduce your issue, but the code above works for me without errors.
![image](https://github.com/Aiving/material-colors/assets/73419411/d5d70daa-91a6-44f4-bb48-2e8776cd0176)

**Comment by subashz** [comment]
Getting similar issue. From debugging it seems
        let wu_result = QuantizerWu::default().quantize(pixels, max_colors, None);
is causing the issue
![screenshot-2024-05-23-17-35-46](https://github.com/Aiving/material-colors/assets/12681896/1a3dead8-b2fc-4f71-bdc4-b9a32219888f)

**Comment by Aiving** [comment]
fixed in e90c351

**Comment by tehrelt** [comment]
have a same problem

```rust
use material_colors::{
    image::{FilterType, ImageReader},
    theme::ThemeBuilder,
};

#[tokio::main]
async fn main() -> Result<(), reqwest::Error> {
    let image = reqwest::get("https://picsum.photos/200/300")
        .await?
        .bytes()
        .await?
        .to_vec();

    let mut data = ImageReader::read(image).expect("failed to read image");

    data.resize(128, 128, FilterType::Nearest);
    let theme = ThemeBuilder::with_source(ImageReader::extract_color(&data)).build();

    Ok(())
}
```

![terminal output](https://github.com/user-attachments/assets/19072c1b-08cb-40da-971b-d49a0aa281ea)
![debug image](https://github.com/user-attachments/assets/1890834a-7c29-481b-8fa9-f3d5308fd529)
